Unable to copy a contentlet having Site/Folder Field

### Parent Issue

_No response_

### Problem Statement

Having a content type with a field Site/Folder(Not Required).
When we try to create a copy of this content, it throws an error 

`Caused by: java.lang.NullPointerException
	at com.dotmarketing.business.DeterministicIdentifierAPIImpl.lambda$deterministicIdSeed$4a664f34$1(DeterministicIdentifierAPIImpl.java:171) ~[dotcms_23.01.7_999999.jar:?]
	at io.vavr.control.Try.of(Try.java:75) ~[vavr-0.10.3.jar:?]
	at com.dotmarketing.business.DeterministicIdentifierAPIImpl.deterministicIdSeed(DeterministicIdentifierAPIImpl.java:169) ~[dotcms_23.01.7_999999.jar:?]
	at com.dotmarketing.business.DeterministicIdentifierAPIImpl.generateDeterministicIdBestEffort(DeterministicIdentifierAPIImpl.java:368) ~[dotcms_23.01.7_999999.jar:?]
	at com.dotmarketing.business.IdentifierAPIImpl.createNew(IdentifierAPIImpl.java:194) ~[dotcms_23.01.7_999999.jar:?]
	at com.dotmarketing.business.IdentifierAPIImpl.createNew(IdentifierAPIImpl.java:138) ~[dotcms_23.01.7_999999.jar:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.addOrUpdateContentletIdentifier(ESContentletAPIImpl.java:5154) ~[dotcms_23.01.7_999999.jar:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.internalCheckin(ESContentletAPIImpl.java:4956) ~[dotcms_23.01.7_999999.jar:?]
	... 22 more`

### Steps to Reproduce

1. Create a content type: Test having a Site/Folder field included  (This is not a required field: Required checkbox is not chosen)

<img width="998" alt="Screenshot 2023-10-18 at 3 45 25 PM" src="https://github.com/dotCMS/core/assets/139247809/354ee618-2ad6-43c1-96d1-b6a88c522add">

2. Create Content Test A for content Type Test (This will have a default value as **System Host / Root Site** assigned to Site/Folder) field 
<img width="1231" alt="Screenshot 2023-10-18 at 2 16 17 PM" src="https://github.com/dotCMS/core/assets/139247809/e479f34d-8ba7-435d-9553-dc51ce331c59">

3. Perform copy action on Test A to create a copy 
<img width="1341" alt="Screenshot 2023-10-18 at 3 47 40 PM" src="https://github.com/dotCMS/core/assets/139247809/4207b454-7322-4b7c-8e06-bd1b0efd2187">

4. This results in an error.
5. When choosing any specific folder under the root (application) for example, the copy function works fine.
<img width="1592" alt="Screenshot 2023-10-18 at 3 47 13 PM" src="https://github.com/dotCMS/core/assets/139247809/0f9e323e-98d8-48d4-9938-6c186fa6cbb8">

Error logs attached from demo : 
[dotcms (24).log](https://github.com/dotCMS/core/files/13033383/dotcms.24.log)


### Acceptance Criteria

Copy of a contentlet should work in case of having a Site/Folder Field included and is optional without throwing any errors 

### dotCMS Version

23.09.7
23.01.7


### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.slack.com/archives/G01N0AU1NUC/p1697650841389619\
https://dotcms.zendesk.com/agent/tickets/113818

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

--- a/dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java
+++ b/dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java
@@ -2774,7 +2774,10 @@ public class ESContentletAPIImplTest extends IntegrationTestBase {
     public void test_copy_contentlet() throws DotDataException, DotSecurityException {
         final List<Field> fields = new ArrayList<>();
         fields.add(new FieldDataGen().name("Title").velocityVarName("title").next());
-        fields.add(new FieldDataGen().type(HostFolderField.class).name("Host").velocityVarName("Host").next());
+        fields.add(new FieldDataGen().type(HostFolderField.class)
+                .name(Host.HOST_VELOCITY_VAR_NAME)
+                .velocityVarName(Host.HOST_VELOCITY_VAR_NAME)
+                .next());
         final ContentType cType = new ContentTypeDataGen()
                 .host(APILocator.systemHost())
                 .fields(fields)

--- a/dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java
+++ b/dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java
@@ -11,6 +11,7 @@ import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertNotEquals;
 import static org.junit.Assert.fail;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
@@ -28,6 +29,7 @@ import com.dotcms.contenttype.model.field.Field;
 import com.dotcms.contenttype.model.field.FieldBuilder;
 import com.dotcms.contenttype.model.field.RelationshipField;
 import com.dotcms.contenttype.model.field.TextField;
+import com.dotcms.contenttype.model.field.HostFolderField;
 import com.dotcms.contenttype.model.type.ContentType;
 import com.dotcms.contenttype.model.type.ContentTypeBuilder;
 import com.dotcms.contenttype.model.type.SimpleContentType;
@@ -2757,6 +2759,33 @@ public class ESContentletAPIImplTest extends IntegrationTestBase {
         } finally {
             APILocator.getExperimentsAPI().end(experiment.id().orElseThrow(), APILocator.systemUser());
         }
+
+    }
+
+    /**
+     * Method to test: {@link ESContentletAPIImpl#copyContentlet(Contentlet, User, boolean)}
+     * Given Scenario:
+     * Unable to copy a contentlet with Host/Folder field. Error is thrown when the field name is "Host"
+     * ExpectedResult: Copy action should execute successfully without null-pointer error.
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void test_copy_contentlet() throws DotDataException, DotSecurityException {
+        final List<Field> fields = new ArrayList<>();
+        fields.add(new FieldDataGen().name("Title").velocityVarName("title").next());
+        fields.add(new FieldDataGen().type(HostFolderField.class).name("Host").velocityVarName("Host").next());
+        final ContentType cType = new ContentTypeDataGen()
+                .host(APILocator.systemHost())
+                .fields(fields)
+                .nextPersisted();
+        final Contentlet contentlet = new ContentletDataGen(cType.id())
+                .host(APILocator.systemHost())
+                .nextPersisted();
+        final Contentlet respCont =  contentletAPI.copyContentlet(contentlet, user, false);
+
+        assertNotEquals(respCont.getIdentifier(), contentlet.getIdentifier());
+        assertEquals(respCont.getHost(), APILocator.systemHost().getIdentifier());
 
     }
 }

--- a/dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java
+++ b/dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java
@@ -9045,7 +9045,7 @@ public class ESContentletAPIImpl implements ContentletAPI {
 
                 //verify if the velocity var name is not Host to avoid override the host
                 if (tempField.getFieldType().equals(Field.FieldType.HOST_OR_FOLDER.toString())
-                        && !"Host".equalsIgnoreCase(tempField.getVelocityVarName())) {
+                        && !Host.HOST_VELOCITY_VAR_NAME.equalsIgnoreCase(tempField.getVelocityVarName())) {
                     if (folder != null || host != null) {
                         newContentlet.setStringProperty(tempField.getVelocityVarName(),
                                 folder != null ? folder.getInode() : host.getIdentifier());

--- a/dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java
+++ b/dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java
@@ -9043,8 +9043,9 @@ public class ESContentletAPIImpl implements ContentletAPI {
                     }
                 }
 
+                //verify if the velocity var name is not Host to avoid override the host
                 if (tempField.getFieldType().equals(Field.FieldType.HOST_OR_FOLDER.toString())
-                        && !tempField.getVelocityVarName().equalsIgnoreCase("host") ) {
+                        && !"Host".equalsIgnoreCase(tempField.getVelocityVarName())) {
                     if (folder != null || host != null) {
                         newContentlet.setStringProperty(tempField.getVelocityVarName(),
                                 folder != null ? folder.getInode() : host.getIdentifier());

--- a/dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java
+++ b/dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java
@@ -9043,7 +9043,8 @@ public class ESContentletAPIImpl implements ContentletAPI {
                     }
                 }
 
-                if (tempField.getFieldType().equals(Field.FieldType.HOST_OR_FOLDER.toString())) {
+                if (tempField.getFieldType().equals(Field.FieldType.HOST_OR_FOLDER.toString())
+                        && !tempField.getVelocityVarName().equalsIgnoreCase("host") ) {
                     if (folder != null || host != null) {
                         newContentlet.setStringProperty(tempField.getVelocityVarName(),
                                 folder != null ? folder.getInode() : host.getIdentifier());

